zaza/openstack/charm_tests/kerberos/setup.py

```diff
@@ -0,0 +1,239 @@
+# Copyright 2020 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Setup for keystone-kerberos tests."""
+
+import logging
+import tempfile
+import zaza.model
+from zaza.openstack.utilities import openstack as openstack_utils
+from zaza.openstack.charm_tests.kerberos import KerberosConfigurationError
+
+
+def get_unit_full_hostname(unit_name):
+    """Retrieve the full hostname of a unit."""
+    for unit in zaza.model.get_units(unit_name):
+        result = zaza.model.run_on_unit(unit.entity_id, 'hostname -f')
+        hostname = result['Stdout'].rstrip()
+    return hostname
+
+
+def add_empty_resource_file_to_keystone_kerberos():
+    """Add an empty resource to keystone kerberos to complete the setup."""
+    logging.info('Attaching an empty keystone keytab to the keystone-kerberos'
+                 ' unit')
+    tmp_file = '/tmp/empty.keytab'
+    with open(tmp_file, 'w'):
+        pass
+
+    zaza.model.attach_resource('keystone-kerberos',
+                               'keystone_keytab',
+                               tmp_file)
+    logging.info('Waiting for keystone-kerberos unit to be active and idle')
+    unit_name = zaza.model.get_units('keystone-kerberos')[0].name
+    zaza.model.block_until_unit_wl_status(unit_name, "active")
+    zaza.model.block_until_all_units_idle()
+
+
+def add_dns_entry(kerberos_hostname="kerberos.testubuntu.com"):
```

Comment:
Right, I agree that it might not be ideal to have to edit this file. 

I think one way to get around that would be to configure the kerberos server with its original hostname, instead of changing it for "kerberos.testubuntu.com". I went this way to make it more obvious that requests towards that server would indeed be for the kerberos server, and to keep the same name every time the test fixture is deployed. 

Otherwise, hostnames of nodes deployed by juju are something like this "juju-75c90b-zaza-2b19a3604ff8-0.project.serverstack". But it would be possible to change the test-fixture to keep its original hostname, and the name resolution should work without editing the /etc/hosts file.